test/metabase/sync/sync_metadata/fields/sync_metadata_test.clj

```diff
@@ -236,3 +240,61 @@
              :json-unfolding    false
              :database-required false
              :database-is-auto-increment false})))))
+
+(deftest base-type-change-will-trigger-fingerprint-and-analyze-test
+  (testing "A base type of a field changes only when the field is dropped then a new field with the name is created (#37047).
+           In this case, we should make sure effective type is set to base type"
+    (is (= [["Field"
+             1
+             {:base_type           :type/Text
+              :effective_type      :type/Text
+              :coercion_strategy   nil
+              :fingerprint_version 0
+              :fingerprint         nil
+              :semantic_type       nil}]]
+           (updates-that-will-be-performed
+            (merge default-metadata
+                   {:id             1
+                    :base-type      :type/Text
+                    :effective-type :type/Text})
+            (merge default-metadata
+                   {:id             1
+                    :base-type      :type/Integer
+                    :effective-type :type/Integer}))))
+
+    (testing "and sync will re-fingerprint and analyze this field"
+      (mt/dataset (mt/dataset-definition "base-type-change"
+                    ["table"
+                     [{:field-name "field"
+                       :base-type  :type/Text}]
+                     [["ngoc@metabase.com"]]])
```

Comment:
This dataset and later changes to the test db will remain after the test finishes, which means this test doesn't run in isolation. You should wrap this test in `mt/with-empty-db` to be completely safe

Reply:
it was wrapped with a try-finally to delete the db afterward, so there should no new db after the test.

Reply:
Sorry, my comment was ambiguous. I'm talking about changes to the actual test database that is created with `mt/dataset`, not the database data in the application db. The finally block only cleans up the database in the application db.